dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHook.java

```diff
@@ -464,25 +494,7 @@ private void checkEventWriteAccess( ValidationErrorReporter reporter, Event even
         checkNotNull( programStage, PROGRAM_STAGE_CANT_BE_NULL );
         checkNotNull( programStage.getProgram(), PROGRAM_CANT_BE_NULL );
 
-        if ( eventOrgUnit == null )
-        {
-            log.warn( "ProgramStageInstance " + event.getUid()
-                + " has no organisation unit assigned, so we skip user validation" );
-        }
-        else if ( isCreatableInSearchScope
-            ? !organisationUnitService.isInUserSearchHierarchyCached( user, eventOrgUnit )
-            : !organisationUnitService.isInUserHierarchyCached( user, eventOrgUnit ) )
-        {
-
-            TrackerErrorReport error = TrackerErrorReport.builder()
-                .uid( event.getUid() )
-                .trackerType( TrackerType.EVENT )
-                .errorCode( TrackerErrorCode.E1000 )
-                .addArg( user )
-                .addArg( eventOrgUnit )
-                .build( bundle );
-            reporter.addError( error );
-        }
+        checkEventOrgUnitWriteAccess( reporter, event, eventOrgUnit, isCreatableInSearchScope, bundle, user );
```

Comment:
I would also group parameters `reporter, bundle` first.

this keeps it familiar since our TrackerValidationHook interface is
`void validate( ValidationErrorReporter report, TrackerImportValidationContext bundle )`